objects/src/accounts/component/template/storage/init_storage_data.rs

```diff
@@ -0,0 +1,35 @@
+use alloc::collections::BTreeMap;
+
+use super::{StoragePlaceholder, StorageValue};
+
+/// Represents the data required to initialize storage entries when instantiating an
+/// [AccountComponent](crate::accounts::AccountComponent).
+#[derive(Clone, Debug, Default)]
+pub struct InitStorageData {
+    /// A mapping of storage placeholder names to their corresponding storage values.
+    storage_placeholders: BTreeMap<StoragePlaceholder, StorageValue>,
+}
+
+impl InitStorageData {
+    /// Creates a new instance of [InitStorageData].
+    ///
+    /// # Parameters
+    ///
+    /// - `entries`: An iterable collection of key-value pairs.
+    pub fn new(entries: impl IntoIterator<Item = (StoragePlaceholder, StorageValue)>) -> Self {
+        InitStorageData {
+            storage_placeholders: entries.into_iter().collect(),
+        }
+    }
+
+    /// Retrieves a reference to the storage placeholders.
+    pub fn storage_placeholders(&self) -> &BTreeMap<StoragePlaceholder, StorageValue> {
+        &self.storage_placeholders
+    }
```

Comment:
nit: maybe just `placeholders()` would be sufficient?

Reply:
Changed